signalhandler_unix.go

```diff
@@ -38,6 +38,7 @@ type SignalHandler struct {
 func NewSignalHandler() *SignalHandler {
 	return &SignalHandler{
 		done:      make(chan struct{}),
+		stop:      make(chan struct{}),
```

Comment:
There's also a missing return in the stop select of handle

Reply:
This was one that I was cleaning up based on the original code - stop was never initiated, and so it panicked on start up. I didn't get around to reviewing whether it was even needed, I can have a look through. It's used later if you decide to register a different manager than the default one

Reply:
yer it looks like the name was refactored and that was incomplete, I may be wrong but that's what it feals like.

Reply:
I agree - will see if I can remove it and get done to do everything

Reply:
This still looks like it needs fixing

Reply:
Yeah I said in the commit comment I'll fix it in the wider refactor

Reply:
as you've made changes to stop / done it would be better to just make that fix here so it keeps the bits together. It should be just a matter of standardising on the one instead of using two.